### DISCUSSION TITLE
Target that runs all the time

### DISCUSSION THREAD

**Discussion by victorhurdugaci** [magefile/mage#385] [Q&A]
Hello,

What is the recommended way to run a target or function before any target? 

The issue I'm trying to solve is that we have some code that needs to run before _all_ targets. I was thinking about `init` but that sounds too hacky and unpredictable... Requiring each target to depend on a _common_ target is also error-prone because one might forget it... Is there an equivalent of `TestMain`?

Cheers

**Comment by ladydascalie** [comment]
I would in fact recommend using `init`.
I've used it in the past for that purpose and have never ran into any issues doing so.
Sometimes I even use it to setup some configuration values for imported targets, to great effect.

**Reply by natefinch** [reply]
Yeah, I don't generally recommend init, but if you legit need something to run before targets, that'll do it.
